src/GraphQl/Type/SchemaBuilder.php

```diff
@@ -201,17 +204,43 @@ private function getResourceFieldConfiguration(string $fieldDescription = null,
             }
 
             $args = [];
-            if ($this->paginationEnabled && !$isInternalGraphqlType && $type->isCollection() && !$input && null === $mutationName) {
-                $args = [
-                    'first' => [
-                        'type' => GraphQLType::int(),
-                        'description' => 'Returns the first n elements from the list.',
-                    ],
-                    'after' => [
-                        'type' => GraphQLType::string(),
-                        'description' => 'Returns the elements in the list that come after the specified cursor.',
-                    ],
-                ];
+            if (!$input && null === $mutationName && !$isInternalGraphqlType && $type->isCollection()) {
+                if ($this->paginationEnabled) {
+                    $args = [
+                        'first' => [
+                            'type' => GraphQLType::int(),
+                            'description' => 'Returns the first n elements from the list.',
+                        ],
+                        'after' => [
+                            'type' => GraphQLType::string(),
+                            'description' => 'Returns the elements in the list that come after the specified cursor.',
+                        ],
+                    ];
+                }
+
+                foreach ($resourceMetadata->getGraphqlAttribute('query', 'filters', [], true) as $filterId) {
+                    if (!$this->filterLocator->has($filterId)) {
+                        continue;
+                    }
+
+                    foreach ($this->filterLocator->get($filterId)->getDescription($resourceClass) as $key => $value) {
+                        $nullable = isset($value['required']) ? !$value['required'] : true;
+                        $filterType = \in_array($value['type'], Type::$builtinTypes, true) ? new Type($value['type'], $nullable) : new Type('object', $nullable, $value['type']);
+                        $graphqlFilterType = $this->convertType($filterType);
+
+                        if ('[]' === $newKey = substr($key, -2)) {
```

Comment:
Shouldn't it be in a dedicated method (in FilterInterface?)? Or maybe a boolean returned by `getDescription` (like `$value['is_property_list']`?

Reply:
Indeed, but has it was done this way for historical reason (and we definitely need this code for existing custom filters), I propose to add the new key in a new PR.